### ISSUE TITLE
`messageBuffer` support

### ISSUE THREAD

**Issue by AlaaZorkane** [davidyaha/graphql-redis-subscriptions#354]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `graphql-redis-subscriptions@2.3.1` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.d.ts b/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.d.ts
index bad3e89..d5a916c 100644
--- a/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.d.ts
+++ b/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.d.ts
@@ -37,5 +37,5 @@ export declare type Trigger = string | Path;
 export declare type TriggerTransform = (trigger: Trigger, channelOptions?: unknown) => string;
 export declare type Reviver = (key: any, value: any) => any;
 export declare type Serializer = (source: any) => string;
-export declare type Deserializer = (source: string) => any;
+export declare type Deserializer = (source: any) => any;
 export {};
diff --git a/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.js b/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.js
index 443893a..6ea54fa 100644
--- a/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.js
+++ b/node_modules/graphql-redis-subscriptions/dist/redis-pubsub.js
@@ -82,8 +82,8 @@ var RedisPubSub = (function () {
                 console.error("No publisher or subscriber instances were provided and the package 'ioredis' wasn't found. Couldn't create Redis clients.");
             }
         }
-        this.redisSubscriber.on('pmessage', this.onMessage.bind(this));
-        this.redisSubscriber.on('message', this.onMessage.bind(this, undefined));
+        this.redisSubscriber.on('pmessageBuffer', this.onMessage.bind(this));
+        this.redisSubscriber.on('messageBuffer', this.onMessage.bind(this, undefined));
         this.subscriptionMap = {};
         this.subsRefsMap = {};
         this.currentSubscriptionId = 0;
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by skabbes** [comment]
This definitely works, but I think it breaks backwards compatibility.  Maybe an option(s) in the constructor to select "buffer mode" or not.

```javascript
new RedisPubSub({
  ...
  messageEventName: 'messageBuffer',
  pmessageEventName: 'pmessageBuffer',
})
```

But that's a little verbose, maybe:

```javascript
new RedisPubSub({
  ...
  useBufferMessages: true,
})
```

**Comment by AlaaZorkane** [comment]
I agree! Sorry I didn't have time to write a thorough api suggestion but that could def work, for context, we're using redis-pubsub with [mqemitter-redis](https://github.com/mcollina/mqemitter-redis) and it's using [msgpack-lite](https://github.com/kawanet/msgpack-lite) internally, when receiving string messages some bytes get lost through casting therefor don't decode properly.

**Comment by davidyaha** [comment]
Fixed in #360.
Thanks @skabbes

**Comment by davidyaha** [comment]
@skabbes Release version 2.4.0 with message buffer support! 🎉 
Just realized we don't have tests for it. Care to add some?
